bqplot/traits.py

```diff
@@ -145,6 +145,8 @@ def _set_value(self, value):
     def set(self, obj, value):
         #TODO: We shouldnt have to overload the set because
         # numpy doesn't support == comparions
+        #TODO: This should also be removed once _buffered_state_diff is
+        # fixed in ipywidgets
```

Comment:
I  don't agree with this comment. The fact that NdArray emits an event on any assignment (even when the new value is equal to the old one) is a bug regardless of the _buffered_state_diff issue in ipywidgets.

This issue only exposed the problem with nd array.